constraint-solver/src/runtime_constant.rs

```diff
@@ -0,0 +1,97 @@
+use std::ops::{Add, AddAssign, Mul, MulAssign, Neg, Sub};
+
+use num_traits::{One, Zero};
+use powdr_number::FieldElement;
+
+use crate::{
+    quadratic_symbolic_expression::QuadraticSymbolicExpressionImpl,
+    range_constraint::RangeConstraint,
+};
+
+/// Represents a run-time constant in the constraint solver.
+/// Any T: FieldElement can represent a run-time constant (which is also a compile-time constant),
+/// but the trait lets us represent run-time constants symbolically as well.
+pub trait RuntimeConstant<V>:
+    Sized
+    + Neg<Output = Self>
+    + Clone
+    + From<Self::FieldType>
+    + Add<Output = Self>
+    + AddAssign<Self>
+    + Sub<Output = Self>
+    + Mul<Output = Self>
+    + MulAssign<Self>
+    + PartialEq
+    + Eq
+    + Zero
+    + One
+    + TryInto<QuadraticSymbolicExpressionImpl<Self, V>>
+{
+    type FieldType: FieldElement;
+
+    /// Creates a run-time constant from a variable.
+    fn from_symbol(symbol: V, rc: RangeConstraint<Self::FieldType>) -> Self;
+    fn try_to_number(&self) -> Option<Self::FieldType>;
+    fn range_constraint(&self) -> RangeConstraint<Self::FieldType>;
```

Comment:
Can you add some documentation? It seems the default implementation of `is_known_nonzero` depends on a certain behaviour here...